extensions/numi/src/index.tsx

```diff
@@ -56,39 +63,72 @@ export default function Command(props: LaunchProps<{ arguments: NumiArguments }>
   };
 
   const queryOnNumi = (q: string | undefined) => {
-    query(q)
-      .then((results) => {
-        if (toast) {
-          toast.hide();
-        }
-        setState((oldState) => ({ ...oldState, results, isLoading: false }));
-
-        handleUpdateHistory(q, results);
-      })
-      .catch((err) => {
-        if (err.message.includes("ECONNREFUSED")) {
-          setApiStatus(false);
-          setState((oldState) => ({ ...oldState, isLoading: false }));
-        }
+    console.log("Querying", q);
+    if (!use_numi_cli) {
+      query(q)
+        .then((results) => {
+          if (toast) {
+            toast.hide();
+          }
+          setState((oldState) => ({ ...oldState, results, isLoading: false }));
 
-        showToast({
-          style: Toast.Style.Failure,
-          title: "Something went wrong",
-          message: "Please make sure Numi is running",
+          handleUpdateHistory(q, results);
         })
-          .then((toast) => setToast)
-          .catch((err) => console.error("Error Creating Toast"));
-      });
+        .catch((err) => {
+          if (err.message.includes("ECONNREFUSED")) {
+            setApiStatus(false);
+            setState((oldState) => ({ ...oldState, isLoading: false }));
+          }
+
+          showToast({
+            style: Toast.Style.Failure,
+            title: "Something went wrong",
+            message: "Please make sure Numi is running",
+          })
+            .then((toast) => setToast)
+            .catch((err) => console.error("Error Creating Toast"));
+        });
+    } else {
+      queryWithNumiCli(q, numi_cli_binary_path)
+        .then((results) => {
+          console.log("Results", results);
+          if (toast) {
+            toast.hide();
+          }
+          setState((oldState) => ({ ...oldState, results, isLoading: false }));
+
+          if (results.length > 0) {
+            handleUpdateHistory(q, results);
+          }
+        })
+        .catch((err) => {
+          showToast({
+            style: Toast.Style.Failure,
+            title: "Something went wrong",
+            message: "make sure Numi CLI is installed and binary path is correct",
+          })
+            .then((toast) => setToast)
+            .catch((err) => console.error("Error Creating Toast"));
+        });
+    }
   };
 
   const checkApiStatus = () => {
-    query("1")
-      .then(() => setApiStatus(true))
-      .catch((err) => {
-        if (err.message.includes("ECONNREFUSED")) {
-          setApiStatus(false);
-        }
-      });
+    if (!use_numi_cli) {
+      query("1")
+        .then(() => setApiStatus(true))
+        .catch((err) => {
+          if (err.message.includes("ECONNREFUSED")) {
+            setApiStatus(false);
+          }
+        });
+    } else {
+      let path: string | undefined = numi_cli_binary_path;
+      if (path.trim() === "") path = undefined;
```

Comment:
Let the `isNumiCliInstalled` handle the error. If there is no bin path, the exec will fail, or check this on the `isNumiCliInstalled` function.
![CleanShot 2023-07-06 at 23 15 07](https://github.com/raycast/extensions/assets/6563162/4df376a2-9957-4825-8b6c-1f7479ef4daa)

Reply:
ok, I have tried to handle all the errors it gets from `exec` it will return `false` value on `isNumiCliInstalled` function. ensure the output of the path check